TTS/tts/models/neuralhmm_tts.py

```diff
@@ -187,15 +187,15 @@ def _format_aux_input(self, aux_input: Dict, default_input_dict):
             }
         )
         if aux_input:
-            return format_aux_input(aux_input, default_input_dict)
+            return format_aux_input(default_input_dict, aux_input.copy())
         return None
 
     @torch.no_grad()
     def inference(
         self,
         text: torch.Tensor,
-        aux_input={"x_lengths": None, "sampling_temp": None, "max_sampling_time": None, "duration_threshold": None},
-    ):  # pylint: disable=dangerous-default-value
+        aux_input=None,
```

Comment:
The way it is written is dangerous: if the function is called without providing `aux_input` as a parameter, this will modify the instance, and the next call will use the modified instance. e.g.

```python
def f(a, l=[]):
   l.append(a)
   print(l)

>>> f(1)
[1]
>>> f(2)
[2, 1]
```

I understand wanting to document the fields in aux_input, but that way is dangerous code (note that you had to disable a linting warning, that's the warning).

Reply:
I looked into how to "correctly" tell people what keys a dictionary should have, and the two options I see are:
1. In the function docstring
2. Using a typing hint called [TypedDict](https://docs.python.org/3/library/typing.html#typing.TypedDict) which also lets automated tooling also know what to expect.

both of these solutions have issues, too. Frankly, I would prefer it to be in the docstring but I don't think it's important once the changes to `format_aux_input()` go through and it doesn't modify things in-place anymore, because then the usage will be practically safe despite being "bad form".